src/Orleans/CodeGeneration/CodeGeneratorManager.cs

```diff
@@ -30,19 +30,26 @@ internal static class CodeGeneratorManager
         /// <summary>
         /// The runtime code generator.
         /// </summary>
-        private static IRuntimeCodeGenerator codeGeneratorInstance;
+        private IRuntimeCodeGenerator codeGeneratorInstance;
 
         /// <summary>
         /// The code generator cache.
         /// </summary>
         private static ICodeGeneratorCache codeGeneratorCacheInstance;
 
+        private readonly IServiceProvider serviceProvider;
+
+        public CodeGeneratorManager(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
```

Comment:
ITypeFactory again.  In this case, it could also be registered in the service provider to allow it to be injected into the code generator manager